openlibrary/templates/books/custom_carousel.html

```diff
@@ -32,6 +32,7 @@
   $ work_key = book.key if book.key.startswith('/work') else book.get('work_key')
   $if lazy:
     $ img_attr = 'data-lazy'
+    $ data_url = 'data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAAEAAAABCAQAAAC1HAwCAAAAC0lEQVR42mNkYAAAAAYAAjCB0C8AAAAASUVORK5CYII='
```

Comment:
Note you are defining data_url here, but not inside the else statement which is why you are getting the render error


```
/openlibrary/openlibrary/templates/books/custom_carousel.html: error in processing template: UnboundLocalError: local variable 'data_url' referenced before assignment (falling back to default template)
```

Reply:
For else statement should I pass an empty string?

Reply:
Have a rethink about how this code works. Perhaps grab a notebook and try to sketch out the different combinations without looking out this code. How would you write this logic in pseudo code?

Before: There was sometimes a src URI, depending on whether the image was lazy and sometimes a data-src attribute. These were mutually exclusive. When the cover_url is non-truthy,  the lazy code is ignored and the div.carousel__item__blankcover is rendered.

Now: There should always be a`src` attribute. The src attribute should either be cover_url or the base64 URI above. When it's the base64 image it should also have a data-src attribute pointing to cover_url. When the cover_url is non-truthy,  the lazy code is ignored and the div.carousel__item__blankcover is rendered.

Reply:
What about an if statement for outputting the data-src attribute (no else) and an if statement for setting the src attribute?